src/views/FarmAuction/hooks/useCurrentFarmAuction.ts

```diff
@@ -0,0 +1,117 @@
+import { useState, useEffect } from 'react'
+import isEqual from 'lodash/isEqual'
+import { useFarmAuctionContract } from 'hooks/useContract'
+import { Auction, ConnectedBidder, Bidder } from 'config/constants/types'
+import { getBidderInfo } from 'config/constants/farmAuctions'
+import useLastUpdated from 'hooks/useLastUpdated'
+import useRefresh from 'hooks/useRefresh'
+import { AUCTION_BIDDERS_TO_FETCH } from 'config'
+import { BIG_ZERO } from 'utils/bigNumber'
+import { sortAuctionBidders, processAuctionData } from '../helpers'
+
+export const useCurrentFarmAuction = (account: string) => {
```

Comment:
Main hook used to get auction data, get bidders, check if user is whitelisted, load whitelisted user data, refresh leaderboard, etc.